samza-core/src/main/scala/org/apache/samza/storage/ContainerStorageManager.java

```diff
@@ -573,6 +576,25 @@ private StorageEngine createStore(String storeName, TaskName taskName, TaskModel
     return this.sideInputStorageManagers.values().stream().collect(Collectors.toSet());
   }
 
+  /**
+   * Registers any CSM created metrics such as side-inputs related metrics, and standby-task related metrics.
+   * @param metricsReporters metrics reporters to use
+   */
+  public void registerMetrics(Map<String, MetricsReporter> metricsReporters) {
```

Comment:
I don't think this is the standard pattern. The right way would be to pass in the shared MetricsRegistry (created in SamzaContainer) to the CSM, and register metrics on that in CSM#register(). That MetricsRegistry is already registered with the MetricsReporters in SamzaContainer. Does this make sense?

Reply:
That's true, but that pattern doesn't work in this case because both the CSM and SamzaContainer need to have a different registry because they both create SystemConsumersMetrics. Using the same registry causes a conflict in the metrics' name. 

We do use this pattern in TaskInstance, where each TaskInstance is given a set of reporters and it registers its metrics there.

Reply:
+1, we would need to change the `SystemConsumerMetrics` to take a configurable group name instead of using the default class name as the group id. I feel that change is much simpler compared to copying over the boiler plate code for metrics registration to CSM.

Reply:
> SystemConsumerMetrics to take a configurable group name

Alerting in ingraphs and auto-scaling rules at LinkedIn are defined based upon the samza generated metric names. Group name is prefix and initial part of the generated metric name. It will be better to factor these use-cases into account and maintain backwards compatibility when changing the metric names.

Reply:
Done. 
It did not require a group-name change, so all existing metrics remain as is.